app/src/main/java/ch/epfl/favo/view/tabs/addFavor/FavorRequestView.java

```diff
@@ -272,6 +273,11 @@ private void confirmUpdatedFavor() {
     CompletableFuture updateFuture = getViewModel().updateFavor(currentFavor);
     updateFuture.thenAccept(o -> showSnackbar(getString(R.string.favor_edit_success_msg)));
     updateFuture.exceptionally(onFailedResult(getView()));
+
+    if (mImageView.getDrawable() != null) {
```

Comment:
Should check if the image changed, otherwise it's redundant to re-upload no?

Reply:
I thought about this too but I'm not sure how to check if the image has been changed. Do you know how I can do that?

Reply:
No real simple way without keeping track of the object state. Once the cache has been implemented, we can compare the current picture with the cached one.

Reply:
I'm going to add this to the backlog for now. This seems like a big enough task to get its own PR.